### PR TITLE
add name for every referee

### PR DESCRIPTION
**Pull request by MishaKulik**
Replacing id with name, because name is more readable and easy.

### CODE REVIEW

File: components/RefereeStatus.js [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
 import { useReferee } from '../db/ref';
 import { timeToString, extractID } from '../helpers';
-
+import Constants from 'expo-constants';
 import { TableWrapper, Row, Cell } from 'react-native-table-component';
+import * as firebase from 'firebase';
 
 export default function RefereeStatus({ id }) {
     let [time] = useReferee(id);
+    firebase
```
Comment (Pavel-Demidyuk) [inline components/RefereeStatus.js]: please move this code to the database controller

Comment (Pavel-Demidyuk) [reply components/RefereeStatus.js]: final result: db.getCompetetions()

File: components/RefereeStatus.js [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
+        });
     return (
         // <Row
         //     // data={[extractID(id), timeToString(time)]}
         //     // style={{ height: 24, backgroundColor: '#fff' }}
         //     // textStyle={{ margin: 6 }}
         // >
         <TableWrapper style={{ flexDirection: 'row', backgroundColor: '#fff' }}>
-            <Cell data={extractID(id)} textStyle={{ margin: 2, color: '#000' }} />
+            <Cell data={global.name} textStyle={{ margin: 2, color: '#000' }} />
```
Comment (Pavel-Demidyuk) [inline components/RefereeStatus.js]: rendering should not start before the db call is finished

File: db/init.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ let db = firebase.database(db);
  */
 function connectToCompetion(pin, isMain = false) {
     console.log('connecting to', pin);
-
+    global.pinPin = pin;
```
Comment (Pavel-Demidyuk) [inline db/init.js]: is there a better way to get pin of the competition?

**Review by Pavel-Demidyuk — APPROVED** [review]

File: db/ref.js [hunk 4]
```diff
@@ -1,16 +1,20 @@
+import random_name from 'node-random-name';
 import Constants from 'expo-constants';
 import { useEffect, useState } from 'react';
-
+import * as firebase from 'firebase';
```
Comment (pdemidyuk-ib) [inline db/ref.js:4]: please remove firebase import here

File: screens/TimerScreen.js [hunk 5]
```diff
@@ -30,7 +33,7 @@ export default function TimerScreen() {
             btnRound={10}
             id={() => (
                 <Text style={{ color: 'gray', textAlign: 'center' }}>
-                    Your ID: {extractID(Constants.deviceId)}{' '}
+                    Your name: {random_name({ seed:extractID(Constants.deviceId), first: true })}{' '}
```
Comment (max8rr8) [inline screens/TimerScreen.js:36]: Not optimal code will slow render, because name is generated on each render

File: db/ref.js [hunk 6]
```diff
@@ -68,4 +72,13 @@ function useReferee(id) {
...
-export { createReferee, useRefs, useReferee };
+function getName(id) {
+  let name;
+     global.connection.child( 'ref/' + id).once('value', snapshot => {
+    name = snapshot.val().name
+  })
+    return name
+
+}
+
```
Comment (max8rr8) [inline db/ref.js:83]: Replace get name with hook 
```javascript
function useName(id){
  let [name, setName] = useState(0)
  useEffect(()=>{
    function c(snapshot) {
      setName(name.val())
    }
    let n = global.connection.child('ref').child(id).child('name')
    n.on('value', c)
    return ()=>n.off('value', c)
  }, [id])
  return name
}
```

File: components/RefereeStatus.js [hunk 7]
```diff
@@ -1,21 +1,30 @@
...
 import { useReferee } from '../db/ref';
 import { timeToString, extractID } from '../helpers';
-
+import Constants from 'expo-constants';
 import { TableWrapper, Row, Cell } from 'react-native-table-component';
+import * as firebase from 'firebase';
+import {getName} from "../db/ref"
 
 export default function RefereeStatus({ id }) {
+    const [name, setName] = useState();
```
Comment (max8rr8) [inline components/RefereeStatus.js:11]: Replace with
```js
const name = useName(id)
```

File: db/ref.js [hunk 8]
```diff
@@ -1,16 +1,20 @@
...
  */
 function createReferee(ref, isMain) {
     ref.child('ref/' + Constants.deviceId.substr(0, 16)).set({
         id: Constants.deviceId,
+        name: random_name({ seed: extractID(Constants.deviceId), first: true }),
         delta: 0,
         running: false,
-        main: isMain
+        main: isMain,
+        session: 1
```
Comment (max8rr8) [inline db/ref.js:17]: Where do we need session

File: package.json [hunk 9]
```diff
@@ -42,8 +44,7 @@
     "react-native-web": "^0.11.4",
     "react-navigation": "^3.11.0",
     "react-screen-orientation": "0.0.4",
-    "react-useinterval": "^1.0.1",
-    "expo-file-system": "~6.0.0"
+    "react-useinterval": "^1.0.1"
```
Comment (max8rr8) [inline package.json:47]: Do we use react-useinterval? If not remove it from our packages

File: screens/TimerScreen.js [hunk 10]
```diff
@@ -1,13 +1,16 @@
...
 import ReferessConnected from '../components/RefereesConnected';
 import { useTimer } from './../db/timer';
 import Constants from 'expo-constants';
 import { extractID } from '../helpers';
 import { Text, View } from 'react-native';
+import random_name from 'node-random-name';
 
 export default function TimerScreen() {
+    const [session, setSession] = useState(0);
+
```
Comment (max8rr8) [inline screens/TimerScreen.js:13]: We dont use session any where

Comment (max8rr8) [inline screens/TimerScreen.js:13]: Add more optimal ode here
```js
const [name, setName] = useState("")
useEffect(()=>{
  setName = random_name({ seed:extractID(Constants.deviceId), first: true })
}, [])
```

File: screens/TimerScreen.js (hunk 5 above)
Comment (max8rr8) [reply screens/TimerScreen.js:36]: Look comment above and replace `random_name({ seed:extractID(Constants.deviceId), first: true })` with `name`

**Review by max8rr8 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**